### ISSUE TITLE
Deploying endpoint with requestParameters returns error. 

### ISSUE THREAD

**Issue by yaochen** [serverless/serverless#366]
Not sure if I'm using it wrong, but it was working when I was testing with JAWS.

The error:

```
Serverless: | /Users/yc/.node/lib/node_modules/serverless/node_modules/bluebird/js/main/promise.js:677
            throw e;
            ^

ServerlessError: No integration defined for method
```

Here's the s-function.json:

```
{
  "functions": {
    "Static-chartInterval": {
      "envVars": [],
      "custom": {
        "excludePatterns": []
      },
      "handler": "modules/static-chart/interval/handler.handler",
      "timeout": 6,
      "memorySize": 1024,
      "endpoints": {
        "api/v2/static-chart/{symbol}/interval": {
          "method": "GET",
          "authorizationType": "none",
          "apiKeyRequired": false,
          "requestParameters": {
            "integration.request.querystring.color": "method.request.querystring.color",
            "integration.request.querystring.height": "method.request.querystring.height",
            "integration.request.querystring.width": "method.request.querystring.width",
            "integration.request.querystring.startDate": "method.request.querystring.startDate",
            "integration.request.querystring.endDate": "method.request.querystring.endDate",
            "integration.request.querystring.header": "method.request.querystring.header"
          },
          "requestTemplates": {
            "application/json": "{ \n\t \"symbol\":\"$input.params('symbol')\", \n\t \"color\":\"$input.params('color')\", \n\t \"height\":\"$input.params('height')\", \n\t \"width\":\"$input.params('width')\", \n\t \"startDate\":\"$input.params('startDate')\", \n\t \"endDate\":\"$input.params('endDate')\", \n\t \"header\":\"$input.params('header')\" \n}"
          },
          "responses": {
            "400": {
              "statusCode": "400"
            },
            "default": {
              "statusCode": "200",
              "responseParameters": {},
              "responseModels": {},
              "responseTemplates": {
                "application/json": ""
              }
            }
          }
        }
      }
    }
  }
}
```

The JAWS requivalent:

```
{
  "lambda": {
    "envVars": [],
    "deploy": true,
    "package": {
      "optimize": {
        "builder": "browserify",
        "minify": true,
        "ignore": [],
        "exclude": [
          "aws-sdk"
        ],
        "includePaths": []
      },
      "excludePatterns": []
    },
    "cloudFormation": {
      "Description": "",
      "Handler": "aws_modules/static/intraday/handler.handler",
      "MemorySize": 256,
      "Runtime": "nodejs",
      "Timeout": 10
    }
  },
  "apiGateway": {
    "deploy": false,
    "cloudFormation": {
      "Type": "AWS",
      "Path": "api/v1/static/intraday/{symbol}",
      "Method": "GET",
      "AuthorizationType": "none",
      "ApiKeyRequired": false,
      "RequestTemplates": {
        "application/json": "{ \n\t \"symbol\":\"$input.params('symbol')\", \n\t \"color\":\"$input.params('color')\", \n\t \"height\":\"$input.params('height')\", \n\t \"width\":\"$input.params('width')\", \n\t \"starttime\":\"$input.params('starttime')\", \n\t \"endtime\":\"$input.params('endtime')\" \n}"
      },
      "RequestParameters": {
        "integration.request.querystring.color": "method.request.querystring.color",
        "integration.request.querystring.height": "method.request.querystring.height",
        "integration.request.querystring.width": "method.request.querystring.width",
        "integration.request.querystring.starttime": "method.request.querystring.starttime",
        "integration.request.querystring.endtime": "method.request.querystring.endtime"
      },
      "Responses": {
        "400": {
          "statusCode": "400"
        },
        "500": {
          "statusCode": "500"
        },
        "default": {
          "statusCode": "200",
          "responseParameters": {},
          "responseModels": {},
          "responseTemplates": {
            "application/json": ""
          }
        }
      }
    }
  }
}
```

**Comment by erikerikson** [comment]
I have a very similar problem.  The putIntegration call to Amazon (around line 460 of ~/lib/actions/EndpointBuildApiGateway.js, which reads "return _this.ApiGateway.putIntegrationPromised(params)") is returning the "error" parameter with content that is something like:
{
	"cause": {
		"message": "Invalid mapping expression specified: Validation Result: warnings : [], errors : [Invalid mapping identifier specified]",
		"code": "BadRequestException",
		"time": "...",
		"requestId": "...",
		"statusCode": 400,
		"retryable": false,
		"retryDelay": 30
	},
	"isOperational": true,
	"code": "BadRequestException",
	"time": "...",
	"requestId": "...",
	"statusCode": 400,
	"retryable": false,
	"retryDelay": 30
}

There are three issues here:
1. The error is swallowed (so I couldn't troubleshoot the problem without altering or debugging the framework code)
2. The error which is actually reported is reported later in the process of deployment when an attempt to deploy the now broken API (the previous integration is wiped out) is made.
2. The mapping is accepted by the console just fine and consists of previously-considered-valid values.

**Comment by base2arthur** [comment]
Same problem too. 

    Serverless: / /Users/amarinis/.npm-packages/lib/node_modules/serverless/node_modules/bluebird/js/main/promise.js:677
            throw e;
            ^
    ServerlessError: No integration defined for method

All worked fine in Jaws and in v0.0.07.

**Comment by ryansb** [comment]
This might be a result of [this commit](https://github.com/serverless/serverless/commit/41f1c264e976a533b62f249ba093267eeb84a2a4). Can you try using a list for your endpoints?

**Comment by mnort9** [comment]
+1. I receive the 'No integration defined for method' error when specifying any requestParameters

**Comment by ac360** [comment]
I'll fix this today.  Sorry for the inconvenience!

**Comment by ac360** [comment]
I believe I found the issue.  A single character was capitalized when it should have been lowercase *sigh*.  This is fixed in `v0.0.13` which will be coming out tonight.

The error swallowing is intentional.  We plan on printing the errors nicely in the console, just haven't gotten there yet.  Very close!

**Comment by erikerikson** [comment]
Many thanks.  You guys have been cranking important work out so thanks for fitting this one in!

**Comment by yaochen** [comment]
@ac360 thanks for the quick fix.

as a temp workaround, u can just change "requestParameters" to "RequestParameters"; or edit the source file now(which i did/then reverted since i didnt want to break something else).

**Comment by ac360** [comment]
Fixed in v0.0.13  Please upgrade and test :)
